test/externalTests/runners/foundry.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+import os
+import re
+import subprocess
+from pathlib import Path
+from shutil import which
+from string import Template
+from typing import Tuple
+from textwrap import dedent
+
+from exttest.common import AVAILABLE_PRESETS
+from exttest.common import settings_from_preset, get_solc_short_version
+from exttest.common import TestConfig, TestRunner
+
+
+def run_forge_command(command: str, env: dict = None):
+    subprocess.run(
+        command.split(),
+        env=env if env is not None else os.environ.copy(),
+        check=True
+    )
+
+
+class FoundryRunner(TestRunner):
+    """Configure and run Foundry-based projects"""
+
+    profile_tmpl = Template(
+        """
+        [profile.${name}]
+        gas_reports = [\"*\"]
+        auto_detect_solc = false
+        solc = \"${solc}\"
+        evm_version = \"${evm_version}\"
+        optimizer = ${optimizer}
+        via_ir = ${via_ir}
+
+        [profile.${name}.optimizer_details]
+        yul = ${yul}
+    """
```

Comment:
Yeah, it works as well. I don't think that it is overkill though, it serves for this purpose as far as I know. But I do agree that it may be better use `format` here, since it can support more formatting options as described in [PEP-498](https://peps.python.org/pep-0498/).